src/Microsoft.ML.Data/TrainCatalog.cs

```diff
@@ -674,6 +674,31 @@ internal RankingTrainers(RankingCatalog catalog)
             var eval = new RankingEvaluator(Environment, options ?? new RankingEvaluatorOptions() { });
             return eval.Evaluate(data, labelColumnName, rowGroupColumnName, scoreColumnName);
         }
+
+        /// <summary>
+        /// Run cross-validation over <paramref name="numberOfFolds"/> folds of <paramref name="data"/>, by fitting <paramref name="estimator"/>,
+        /// and respecting <paramref name="samplingKeyColumnName"/> if provided.
+        /// Then evaluate each sub-model against <paramref name="labelColumnName"/> and return metrics.
+        /// </summary>
+        /// <param name="data">The data to run cross-validation on.</param>
+        /// <param name="estimator">The estimator to fit.</param>
+        /// <param name="numberOfFolds">Number of cross-validation folds.</param>
+        /// <param name="labelColumnName">The label column (for evaluation).</param>
+        /// <param name="samplingKeyColumnName">Name of a column to use for grouping rows. If two examples share the same value of the <paramref name="samplingKeyColumnName"/>,
+        /// they are guaranteed to appear in the same subset (train or test). This can be used to ensure no label leakage from the train to the test set.
+        /// If <see langword="null"/> no row grouping will be performed.</param>
+        /// <param name="rowGroupColumnName">The name of the groupId column in <paramref name="data"/>.</param>
+        /// <param name="seed">Seed for the random number generator used to select rows for cross-validation folds.</param>
+        /// <returns>Per-fold results: metrics, models, scored datasets.</returns>
+        public IReadOnlyList<CrossValidationResult<RankingMetrics>> CrossValidate(
+            IDataView data, IEstimator<ITransformer> estimator, int numberOfFolds = 5, string labelColumnName = DefaultColumnNames.Label,
+            string samplingKeyColumnName = DefaultColumnNames.GroupId, string rowGroupColumnName = DefaultColumnNames.GroupId, int ? seed = null)
```

Comment:
This fixes https://github.com/dotnet/machinelearning/issues/2685 right? Can you please update your PR description to mention this. Thanks!

Reply:
BTW I've asked @yaeldekel on an email if this API was never added because of [https://github.com/dotnet/machinelearning/issues/3711](https://github.com/dotnet/machinelearning/issues/3711) ?? It's still odd to me that this API wasn't added from the beginning, specially now that with your PR it seems it would have been a straightforward addition. So I wonder if we're missing something

---
In reply to: [446444488](https://github.com/dotnet/machinelearning/pull/5263#discussion_r446444488) [](ancestors = 446444488)

Reply:
yup, @antoniovs1029 can you cc me in that?